src/core/PathUtil.cpp

```diff
@@ -131,28 +136,37 @@ namespace PathUtil
 
 
 
-	QString toAbsolute(const QString & input)
+	QString toAbsolute(const QString & input, bool* ok /* = nullptr*/)
 	{
 		//First, do no harm to absolute paths
 		QFileInfo inputFileInfo = QFileInfo(input);
-		if (inputFileInfo.isAbsolute()) { return input; }
+		if (inputFileInfo.isAbsolute())
+		{
+			if (ok) { *ok = true; }
+			return input;
+		}
 		//Next, handle old relative paths with no prefix
 		QString upgraded = input.contains(":") ? input : oldRelativeUpgrade(input);
 
 		Base base = baseLookup(upgraded);
-		return baseLocation(base) + upgraded.remove(0, basePrefix(base).length());
+		return baseLocation(base, ok) + upgraded.remove(0, basePrefix(base).length());
 	}
 
 	QString relativeOrAbsolute(const QString & input, const Base base)
 	{
 		if (input.isEmpty()) { return input; }
 		QString absolutePath = toAbsolute(input);
 		if (base == Base::Absolute) { return absolutePath; }
-		QString relativePath = baseQDir(base).relativeFilePath(absolutePath);
-		return relativePath.startsWith("..") ? absolutePath : relativePath;
+		bool ok;
+		QString relativePath = baseQDir(base, &ok).relativeFilePath(absolutePath);
+		// Return the relative path if it didn't result in a path starting with ..
+		// and the baseQDir was resolved properly
+		return (relativePath.startsWith("..") || !ok)
+			? absolutePath
+			: relativePath;
 	}
 
-	QString toShortestRelative(const QString & input)
+	QString toShortestRelative(const QString & input, bool allowLocal /* = false*/)
```

Comment:
I'm a bit confused, why are the default arguments commented out?

Reply:
If they're defined in the header this will cause an error saying they're already defaulted or something like that.

Reply:
Oh yeah, I totally missed that they're defined there. Thanks!